src/make.ts

```diff
@@ -969,10 +978,46 @@ function getRelevantConfigStats(stats: any): ConfigureSubphaseStatusItem[] {
 // (thus disappearing from the log with commands) will still have IntelliSense loaded
 // until the next clean configure.
 export async function configure(triggeredBy: TriggeredBy, updateTargets: boolean = true): Promise<number> {
-  // Mark that this workspace had at least one attempt at configuring, before any chance of early return,
-  // to accurately identify whether this project configured successfully out of the box or not.
-  let ranConfigureInCodebaseLifetime: boolean =
-    extension.getState().ranConfigureInCodebaseLifetime;
+  // Before running the --dry-run type of configure for the first time, even in a trusted workspace,
+  // notify the user that it is possible some code to be executed even under "--dry-run" mode.
+  // Note that this is a state variable and can be reseted via the command resetState, causing the popup to show again
+  // even if a successfull --dry-run configure has been run in the past.
+  let ranDryRunInCodebaseLifetime: boolean = extension.getState().ranDryRunInCodebaseLifetime;
+  if (!ranDryRunInCodebaseLifetime && !configuration.getBuildLog()) {
+    // The window popup message should be concise but more logging can be useful in the output channel.
+    logger.message("The Makefile Tools extension process of configuring your project is about to run 'make --dry-run' in order to parse the output for useful information. " +
+                   "This is needed to calculate accurate IntelliSense and targets information. " +
+                   "Although in general 'make --dry-run' only lists (without executing) the operations 'make' would do in the current context, " +
+                   "it is still possible some code to be executed, like $(shell) syntax in the makefile or recursive invocations of the $(MAKE) variable.");
+    logger.message("If you don't feel comfortable allowing this configure process and 'make --dry-run' to be invoked by the extension, " +
+                   "you can chose a recent full, clean, verbose and up-to-date build log as an alternative, via the setting 'makefile.buildLog'. ");
+    // Also, show the output channel for that message to be visible.
+    logger.showOutputChannel();
+    let yesButton: string = localize("yes.dont.show.again", "Yes (don't show again)");
+    let noButton: string = localize("no", "No");
+    const chosen: vscode.MessageItem | undefined = await vscode.window.showErrorMessage<vscode.MessageItem>("Configuring project. Code can still execute in --dry-run mode. Do you want to continue?",
```

Comment:
Please make sure to localize the necessary strings. It's not always required to localize things going to the logging output, we should follow the precedent we've set in this extension (I don't think we localize), but the popup should certainly be localized.